platforms/jvm/plugins-application/src/main/java/org/gradle/jvm/application/tasks/CreateStartScripts.java

```diff
@@ -147,85 +157,66 @@ protected JavaModuleDetector getJavaModuleDetector() {
     /**
      * The environment variable to use to provide additional options to the JVM.
      */
-    @Nullable
     @Optional
     @Input
-    public String getOptsEnvironmentVar() {
-        if (GUtil.isTrue(optsEnvironmentVar)) {
-            return optsEnvironmentVar;
-        }
-
-        if (!GUtil.isTrue(getApplicationName())) {
-            return null;
-        }
-
-        return GUtil.toConstant(getApplicationName()) + "_OPTS";
+    @ReplacesEagerProperty
+    public Property<String> getOptsEnvironmentVar() {
+        return optsEnvironmentVar;
     }
```

Comment:
❓ What about this behavior here? I think we should reproduce it via a bytecode upgrade adapter.

Reply:
Ah, sorry, didn't see the convention. They are not identical, as truthy/falsy is now represented by present/absent values, which might have some subtle differences. But I think if we document the change it should be an acceptable breakage.

Where will we document such behavior changes? Should we already start the doc?

Or: can we instead document it in-place, perhaps in Javadoc, using a custom tag maybe? Can we collect such custom tags and produce some summary doc?

Reply:
Let's add a `TODO` for now to make sure we don't forget to document this.